app/controllers/home_controller.rb

```diff
@@ -1,5 +1,125 @@
+
+
+  def getAvgTotal
+    avg = Quiz.where(teacher:@id).average(:score)
+    if avg.nil?
+      avg = "No quizzes taken!"
+    else 
+      avg = avg.round(2)
+    end
+
+    strk = Quiz.where(teacher:@id).average(:longest_streak)
+    if strk.nil?
+      strk = "No quizzes taken!"
+    else
+      strk = strk.round(2)
+    end
+
+    return [avg, strk]
+  end
+  helper_method :getAvgTotal
+
+  def getAvgRecent
+    recentFall = Course.where(teacher:@id).where("semester like ?", "%#{"Fall"}%").order(semester: :desc).first
+    fallYear = (recentFall.semester[-4..-1] || "0").to_i
+    recentSpring = Course.where(teacher:@id).where("semester like ?", "%#{"Spring"}%").order(semester: :desc).first
+    springYear = (recentSpring.semester[-4..-1] || "0").to_i
+   
+    semester = ""
+    if fallYear > springYear
+      semester = recentFall.semester
+    else
+      semester = recentSpring.semester
+    end
+   
+    courses = Course.select(:id).where(teacher:@id).where("semester like ?", "%#{semester}%")
+    
+    avg = Quiz.where(teacher:@id,id:courses).average(:score)
+    if avg.nil?
+      avg = "No quizzes taken!"
+    else 
+      avg = avg.round(2)
+    end
+
+    strk = Quiz.where(teacher:@id,id:courses).average(:longest_streak)
+    if strk.nil?
+      strk = "No quizzes taken!"
+    else
+      strk = strk.round(2)
+    end
+    
+    return [semester, avg, strk]
+  end
+  helper_method :getAvgRecent
+
+  def studentInfo
+    qID = Quiz.where(teacher:@id).pluck(:id)
+    atm = Qroster.where(quiz_id:qID,correct_resp:true).count(:attempts)
+
+    best = Qroster.where(quiz_id:qID,correct_resp:true).group(:student_id).select(:student_id, "(SUM(CAST(1 AS Float) / CAST(attempts as Float)*100.00) / #{atm}) AS wscore", "(SUM(CAST(1 AS Float) / CAST(attempts as Float)*100.00)/COUNT(attempts)) AS score").order("wscore DESC").first
+    name = Student.where(id:best.student_id,teacher: @id).pick(:firstname, :lastname)
+    bestName = name[0] + " " + name[1]
+    bestInfo = "#{bestName} (#{best.score.round(2)}%)"
+
+
+    worst = Qroster.where(quiz_id:qID,correct_resp:true).group(:student_id).select(:student_id, "(SUM(CAST(1 AS Float) / CAST(attempts as Float)*100.00) / #{atm}) AS wscore", "(SUM(CAST(1 AS Float) / CAST(attempts as Float)*100.00)/COUNT(attempts)) AS score").order("wscore ASC").first
+    name = Student.where(id:worst.student_id,teacher: @id).pick(:firstname, :lastname)
+    worstName = name[0] + " " + name[1]
+    worstInfo = "#{worstName} (#{worst.score.round(2)}%)"
+    
+    ids = Qroster.where(quiz_id:qID,correct_resp:true).select(:student_id).distinct.pluck(:student_id)
+    logger.info("IDS: #{ids}")
+    maxCnt = 0
+    maxID = 0
+    minCnt = Float::INFINITY
+    minID = 0
+    ids.each do |i|
+      tmpCnt = 0
+      qr = Qroster.where(quiz_id:qID,student_id:i,correct_resp:true).order(updated_at: :desc).pluck(:attempts)
+      logger.info("HIGHEST: #{qr}")
+      qr.each do |att|
+        if att == 1
```

Comment:
why is att set to 1? I think we should just remove this and also I think we should remove maxCnt and minCnt outside `qr.each do |att|` loop as we should count the attempts and then calculate max and min count outside the loop for each student id

Reply:
I still the same about this. let's talk on this.

Reply:
Ask prof about adding a tolerance

Reply:
Prof said 0 tolerance